File: AppURLMS/app/src/androidTest/java/ca/mcgill/ecse321/appurlms/PersistenceTest.java
package ca.mcgill.ecse321.appurlms;

import android.support.test.runner.AndroidJUnit4;
import org.junit.*;
import org.junit.runner.RunWith;

import static junit.framework.Assert.fail;
import static org.junit.Assert.assertEquals;

import java.sql.Date;

import ca.mcgill.ecse321.urlms.model.*;
import ca.mcgill.ecse321.urlms.persistence.PersistenceXStream;

@RunWith(AndroidJUnit4.class)
public class PersistenceTest {
    // Tests values
    private static String testEmail ="director@urlms.ca";
    private static String testStaffEmail ="staff@urlms.ca";
    private static String testPassword ="password";
    private static String testDirName ="Director";
    private static String testStaffName ="Staff";

    private URLMS urlms;

    @Before
    public void setUp() {
        urlms = new URLMS();

        // Create participants
        Director dr = new Director(testEmail, testPassword, testDirName, urlms);
        urlms.addLaboratory("LabOne", "Test", new Date(2017, 10, 10), new Date(2017, 10, 10), true, dr);
        Staff staffMember = new Staff(testStaffEmail, testPassword, testStaffName);
        urlms.getLaboratory(0).addStaff(staffMember);

        // Create data file
        PersistenceXStream.initializeURLMS("/data.xml");
        PersistenceXStream.saveToXMLwithXStream(urlms);
    }

    @After
    public void tearDown() {
        urlms.delete();
    }

    @Test
    public void test() {
        // Clear the model in memory
        urlms.delete();
        assertEquals(false, urlms.hasDirectors());
        assertEquals(false, urlms.hasLaboratories());

        // Load model
        urlms = (URLMS) PersistenceXStream.loadFromXMLwithXStream();
        if (urlms == null)
            fail("Could not load file.");

        // Check participants (Staff and director)
        assertEquals(true, urlms.hasDirectors());
        assertEquals(true, urlms.hasLaboratories());
        assertEquals(true, urlms.getLaboratory(0).hasStaffs());
        assertEquals(testEmail, urlms.getDirector(0).getEmail());
        assertEquals(testPassword, urlms.getDirector(0).getPassword());
        assertEquals(testDirName, urlms.getDirector(0).getName());
        assertEquals(testStaffEmail, urlms.getLaboratory(0).getStaff(0).getEmail());
        assertEquals(testPassword, urlms.getLaboratory(0).getStaff(0).getPassword());
        assertEquals(testStaffName, urlms.getLaboratory(0).getStaff(0).getName());
    }
}

File: WebURLMS/test/PersistenceTest.php
<?php
foreach(glob(dirname(__FILE__) . '/../model/*.php') as $file){
    require_once $file;
}

require_once dirname(__FILE__) . '/../controller/Controller.php';

class PersistenceTest extends PHPUnit\Framework\TestCase
{
    private $pm;
    private $urlms;
    
    private $testEmail ="director@urlms.ca";
    private $testStaffEmail ="staff@urlms.ca";
    private $testPassword ="password";
    private $testDirName ="Director";
    private $testStaffName ="Staff";
    
    public function setUp()
    {
        $this->pm = new PersistenceManager();
        
        $this->urlms = URLMS::getInstance();
        
        // Create participants
        $dr = new Director($this->testEmail, $this->testPassword, $this->testDirName, $this->urlms);
        $lab = $this->urlms->addLaboratoryVia("LabOne", "Test", new DateTime("now"), true, $dr);
        $staffMember = new Staff($this->testStaffEmail, $this->testPassword, $this->testStaffName, StaffRole::ResearchAssistant, [$lab]);
        // Create data file
        $this->pm->writeDataToStore($this->urlms);
    }
    
    public function tearDown()
    {
        $this->urlms->delete();
    }
    
    public function testPersistence()
    {
        //Delete current model
        $this->urlms->delete();

        //Assure that nothing is there
        $this->assertEquals(false, $this->urlms->hasDirectors());
        $this->assertEquals(false, $this->urlms->hasLaboratories());
        
        // Load model
        $this->urlms = $this->pm->loadDataFromStore();
            
        // Check participants (Staff and director)
        $this->assertEquals(true, $this->urlms->hasDirectors());
        $this->assertEquals(true, $this->urlms->hasLaboratories());
        $this->assertEquals(true, $this->urlms->getLaboratory_index(0)->hasStaffs());
        $this->assertEquals($this->testEmail, $this->urlms->getDirector_index(0)->getEmail());
        $this->assertEquals($this->testPassword, $this->urlms->getDirector_index(0)->getPassword());
        $this->assertEquals($this->testDirName, $this->urlms->getDirector_index(0)->getName());
        $this->assertEquals($this->testStaffEmail, $this->urlms->getLaboratory_index(0)->getStaff_index(0)->getEmail());
        $this->assertEquals($this->testPassword, $this->urlms->getLaboratory_index(0)->getStaff_index(0)->getPassword());
        $this->assertEquals($this->testStaffName, $this->urlms->getLaboratory_index(0)->getStaff_index(0)->getName());
    }
    
}
?>